iroha/src/tx.rs

```diff
@@ -280,6 +283,63 @@ impl From<AcceptedTransaction> for Transaction {
     }
 }
 
+impl From<VersionedValidTransaction> for Transaction {
+    fn from(tx: VersionedValidTransaction) -> Self {
+        let tx: ValidTransaction = tx.into_inner_v1();
+        tx.into()
```

Comment:
This conversion won't be true for too long, because `into_inner_v1` api will change.

In general there should be conversions between versioned types like
`From<VersionedValidTransaction> for VersionedTransaction`
etc

Reply:
IMO it should be removed

Reply:
Agree. There are at least 2 places in the same file that also use `into_inner_v1()` conversion. Should they will be changed too?

Reply:
I guess no. You should use it in places where you would anyway match on inner structure.
For now Versioned enums have only 1 variant, so sometimes it makes sense to get inner structure, but in many times it is not the case (like here). Api should be designed in such way that it can be replaced in future with match for all versions of specific type